app/src/main/java/co/nimblehq/alpine/sample/CameraCaptureActivity.kt

```diff
@@ -85,11 +112,89 @@ class CameraCaptureActivity : ComponentActivity() {
         )
         // Attach the viewFinder's surface provider to Preview use case
         preview?.setSurfaceProvider(binding.pvCameraCaptureViewFinder.surfaceProvider)
+    }
+
+    private fun bindViewEvents() {
+        observePhotoCapture()
+        binding.ibCameraCapture.setOnClickListener {
+            loadingDialog.show()
+            imageCapture?.let {
+                val photoFile = createFile(outputDirectory)
+                val outputOptions = ImageCapture.OutputFileOptions.Builder(photoFile)
+                    .build()
+
+                // Setup image capture listener which is triggered after photo has been taken
+                it.takePicture(
+                    outputOptions,
+                    cameraExecutor,
+                    object : ImageCapture.OnImageSavedCallback {
+                        override fun onError(exc: ImageCaptureException) {
+                            showErrorMessage()
+                        }
+
+                        override fun onImageSaved(output: ImageCapture.OutputFileResults) {
+                            photoCaptured.value = photoFile
```

Comment:
Just realised, we could just directly call `processImage` here instead of passing it to an observable 🤔 What do you think? 😄 
```suggestion
                            processImage(photoFile)
```